python/lsst/ap/pipe/ap_pipe.py

```diff
@@ -21,356 +21,327 @@
+            The output of the task assigned to ``config.differencer`
+            (by default, `lsst.pipe.tasks.ImageDifferenceTask.run`)
+        associator : `lsst.pipe.base.Struct` or `None`
+            The output of the task assigned to ``config.associator`
+            (by default, `lsst.ap.association.AssociationTask.run`)
+        """
+        # Ensure that templateIds make it through basic data reduction
+        # TODO: treat as independent jobs (may need SuperTask framework?)
+        if templateIds is not None:
+            for templateId in templateIds:
+                rawTemplateRef = _siblingRef(rawRef, "raw", templateId)
+                calexpTemplateRef = _siblingRef(calexpRef, "calexp", templateId)
+                if "ccdProcessor" not in reuse or not calexpTemplateRef.datasetExists("calexp", write=True):
+                    self.runProcessCcd(rawTemplateRef)
+
+        if "ccdProcessor" in reuse and calexpRef.datasetExists("calexp", write=True):
+            self.log.warn("ProcessCcd has already been run for {0}, skipping...".format(rawRef.dataId))
+            processResults = None
+        else:
+            processResults = self.runProcessCcd(rawRef)
+
+        diffType = self.config.differencer.coaddName
+        if "differencer" in reuse and calexpRef.datasetExists(diffType + "Diff_diaSrc", write=True):
+            self.log.warn("DiffIm has already been run for {0}, skipping...".format(calexpRef.dataId))
+            diffImResults = None
+        else:
+            diffImResults = self.runDiffIm(calexpRef, templateIds)
+
+        # No reasonable way to check if Association finished successfully
+        associationResults = self.runAssociation(calexpRef)
+
+        return pipeBase.Struct(
+            fullMetadata = self.getFullMetadata(),
+            l1Database = associationResults.l1Database,
+            ccdProcessor = processResults.taskResults if processResults else None,
+            differencer = diffImResults.taskResults if diffImResults else None,
+            associator = associationResults.taskResults if associationResults else None
+        )
+
+    @pipeBase.timeMethod
+    def runProcessCcd(self, sensorRef):
+        """Perform ISR with ingested images and calibrations via processCcd.
+
+        The output repository associated with ``sensorRef`` will be populated with the
+        usual post-ISR data (bkgd, calexp, icExp, icSrc, postISR).
+
+        Parameters
+        ----------
+        sensorRef : `lsst.daf.persistence.ButlerDataRef`
+            Data reference for raw data.
+
+        Returns
+        -------
+        fullMetadata : `lsst.daf.base.PropertySet`
+            The metadata produced by the Task. Intended as a transitional API
+            for ap_verify, and may be removed later.
+        taskResults : `lsst.pipe.base.Struct`
+            The output of the task assigned to ``config.ccdProcessor`
+            (by default, `lsst.pipe.tasks.ProcessCcdTask.run`)
```

Comment:
It actually returns an `lsst.pipe.base.Struct` containing the specified named fields. I'm afraid you have omitted any mention of `Struct` here and in most or all of the other `Returns` documentation for this class.